internal/datanode/channel_meta.go

```diff
@@ -294,13 +311,68 @@ func (c *ChannelMeta) filterSegments(partitionID UniqueID) []*Segment {
 }
 
 func (c *ChannelMeta) InitPKstats(ctx context.Context, s *Segment, statsBinlogs []*datapb.FieldBinlog, ts Timestamp) error {
+	if Params.DataNodeCfg.SkipBFStatsLoad {
+		// mark segment lazy loading
+		s.setLoadingLazy(true)
+		c.submitLoadStatsTask(s, statsBinlogs, ts)
+		return nil
+	}
+	return c.initPKstats(ctx, s, statsBinlogs, ts)
+}
+
+func (c *ChannelMeta) submitLoadStatsTask(s *Segment, statsBinlogs []*datapb.FieldBinlog, ts Timestamp) {
+	log := log.Ctx(context.TODO()).With(
+		zap.Int64("segmentID", s.segmentID),
+		zap.Int64("collectionID", s.collectionID),
+	)
+	if c.closed.Load() {
+		// stop retry and resubmit if channel meta closed
+		return
+	}
+	// do submitting in a goroutine in case of task pool is full
+	go func() {
+		c.workerPool.Submit(func() (any, error) {
+			stats, err := c.loadStats(context.Background(), s.segmentID, s.collectionID, statsBinlogs, ts)
+			if err != nil {
+				// TODO if not retryable, add rebuild statslog logic
+				log.Warn("failed to lazy load statslog for segment", zap.Error(err))
+				if c.retryableLoadError(err) {
```

Comment:
should we take of close here?

Reply:
The `closed` check  is at the beginning of this resubmit function